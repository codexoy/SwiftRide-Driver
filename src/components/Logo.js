import React from "react"
import { View, StyleSheet, Image, Text } from "react-native"

export default class Logo extends React.Component {
	constructor(props) {
		super(props)
		this.sizes = {
			small: {
				width: 100,
				height: 80,
				fontSize: 24
			},
			medium: {
				width: 140,
				height: 112,
				fontSize: 40
			},
			large: {
				width: 196,
				height: 156,
				fontSize: 64
			}
		}

		this.state = { ...this.sizes[this.props.size] }
	}
	render() {
		return (
			<View style={{ ...styles.column }}>
				<Image
					source={require("../../assets/images/logo.png")}
					style={{
						width: this.state.width,
						height: this.state.height,
						resizeMode: "contain"
					}}
				/>
				<Text style={{ ...styles.text, fontSize: this.state.fontSize }}>
					Dryvo
				</Text>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	text: {
		fontFamily: "Montserrat-Bold",
		color: "#fff"
	},
	column: {
		flexDirection: "column",
		alignItems: "center"
	}
})
