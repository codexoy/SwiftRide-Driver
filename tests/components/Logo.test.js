import React from "react"

import renderer from "react-test-renderer"

import Logo from "../../src/components/Logo"

jest.useFakeTimers()
describe("Logo", () => {
	test("view renders correctly", () => {
		const tree = renderer.create(<Logo size="large" />).toJSON()
		expect(tree).toMatchSnapshot()
	})
})
