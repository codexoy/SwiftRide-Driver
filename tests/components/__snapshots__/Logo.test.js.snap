// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo view renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/logo.png",
      }
    }
    style={
      Object {
        "height": 156,
        "resizeMode": "contain",
        "width": 196,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 64,
      }
    }
  >
    Dryvo
  </Text>
</View>
`;
